modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -484,6 +505,31 @@ private JSONArray _getFiltersJSONArray(
 					);
 				}
 
+				String clientExtensionERC = MapUtil.getString(
+					properties, "clientExtensionERC");
+
+				if (Validator.isNotNull(clientExtensionERC)) {
+					ThemeDisplay themeDisplay =
+						(ThemeDisplay)httpServletRequest.getAttribute(
+							WebKeys.THEME_DISPLAY);
+
+					FDSFilterCET fdsFilterCET =
+						(FDSFilterCET)_cetManager.getCET(
+							themeDisplay.getCompanyId(), clientExtensionERC);
+
```

Comment:
If there is no filterCET deployed at the time of rendering the fragment, we'll get ǹull`value for `fdsFilterCET`, then logic shall gracefully fail. Most likely, this means:

- Show a warning in the server log 
- Do not send this filter to the UI

This means that the rendered FDS won't be aware that there's a missing filter. I think this is fine

Reply:
Is this something that we have all aligned about? When I say all I mean all teams developing DXP.

Because I think we should decide if we want to be lenient or fail-fast with failing CXs and keep it as a product idiosincracy. Otherwise we run the risk of having disparate criteria which will probably lead to user frustration...

Reply:
I've incorporated this feedback in https://github.com/liferay-frontend/liferay-portal/pull/3688/commits/8da534df7fa7b3af7529c781a2d8e069fa8a7808